src/main/webapp/index.jsp

```diff
@@ -4,22 +4,83 @@
 <!-- [START_EXCLUDE] -->
 <!-- [END_EXCLUDE] -->
 <html>
-<head>
-  <link href='//fonts.googleapis.com/css?family=Marmelad' rel='stylesheet' type='text/css'>
-  <title>Hello App Engine Flexible</title>
-</head>
-<body>
-    <h1>Hello App Engine -- Flexible!</h1>
-
-  <p>This is <%= HelloAppEngine.getInfo() %>.</p>
-  <table>
-    <tr>
-      <td colspan="2" style="font-weight:bold;">Available Servlets:</td>
-    </tr>
-    <tr>
-      <td><a href='/hello'>The servlet</a></td>
-    </tr>
-  </table>
-
-</body>
+    <head>
+        <script src="https://cdn.firebase.com/libs/firebaseui/3.5.2/firebaseui.js"></script>
+        <link type="text/css" rel="stylesheet" href="https://cdn.firebase.com/libs/firebaseui/3.5.2/firebaseui.css" />
+        <link href='//fonts.googleapis.com/css?family=Marmelad' rel='stylesheet' type='text/css'>
+        <title>Hello App Engine Flexible</title>
+    </head>
+    <body>
+        <h1>Hello App Engine -- Flexible!</h1>
+
+        <p>This is <%= HelloAppEngine.getInfo() %>.</p>
+        <table>
+            <tr>
+                <td colspan="2" style="font-weight:bold;">Available Servlets:</td>
+            </tr>
+            <tr>
+                <td><a href='/hello'>The servlet</a></td>
+            </tr>
+        </table>
+
+        <div id="firebaseui-auth-container"></div>
+        <div id="loader">Loading...</div>
+
+        <!-- The core Firebase JS SDK is always required and must be listed first -->
+        <script src="https://www.gstatic.com/firebasejs/7.17.1/firebase-app.js"></script>
+
+        <!-- TODO: Add SDKs for Firebase products that you want to use
+             https://firebase.google.com/docs/web/setup#available-libraries -->
+        <script src="https://www.gstatic.com/firebasejs/7.17.1/firebase-auth.js"></script>
+
+        <script>
```

Comment:
Not sure if this works in this context but rather than having script tags with js code in the body, maybe put all of this in a .js file and tag it as a script in the head? If the intention is to have everything run after the html is loaded, you can add a onload=function() property in the body tag and then call these functions. Only suggesting this because, generally, it's good practice to keep js scripts and html code separate. Other than that, everything looks good!